src/View/Form/AbstractContext.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Cake\View\Form;
+
+use Cake\Network\Request;
+
+/**
+ * Provides common functionality for ContextInterface implementations.
+ */
+abstract class AbstractContext implements ContextInterface
+{
+    /**
+     * The request object.
+     *
+     * @var \Cake\Network\Request
+     */
+    protected $_request;
+
+    /**
+     * The request type of the attached form.
+     *
+     * @var string
+     */
+    protected $_requestType;
+
+    /**
+     * Constructor.
+     *
+     * @param \Cake\Network\Request $request The request object.
+     */
+    protected function __construct(Request $request)
+    {
+        $this->_request = $request;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setRequestType($requestType)
+    {
+        $this->_requestType = $requestType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function val($field)
```

Comment:
The change above would make it easy to do PRG refilling a form that would again post. PRG is an example where the URL and the Form itself don't always redirect only to itself. Another example: It also allows to setup form default values easily though query strings in case you want to send around prepared form URLs etc. There are probably more benefits by going a more HTTPish way and that is treating body and query for any verb.

**I'd love to see that Forms behave as good as HTTP citizens can, treating body and query for all verbs - just in difference precedence, isn't that a good idea?** 

Not without reason I suggested (on IRC) to add an inject hashkey to FormHelper::create() to specify explicitly and independent of its type hashkey which kind of data gets injected automagically into its form field values.

What would be the flip side of the coin of doing the change like I suggested above?